js/base-chat.js

```diff
@@ -0,0 +1,617 @@
+		} else if (msg.type === "error") {
+			d20plus.chat.localHistory.pop();
+		}
+		if (msg.type === "whisper") {
+			if (params[1].playerid === d20_player_id) {
+				d20plus.chat.lastRespondent = params[1].target_name;
+			} else if (params[1].target.includes(d20_player_id)) {
+				d20plus.chat.lastRespondent = d20.Campaign.players.get(params[1].playerid)?.attributes.displayname;
+			}
+		}
+		if (d20plus.chat.mtms?.await && params[1]) {
+			if (params[1].playerid === d20_player_id
+				|| params[1].type === "error") d20plus.chat.mtms.success = true;
+		}
+		if (d20plus.cfg.getOrDefault("chat", "languages") && msg.listenerid?.language) {
+			const speech = msg.listenerid;
+			const is_current_player = msg.playerid === d20_player_id;
+			const knows_language = hasLanguageProfeciency(speech.languageid);
+			const translated = speech.message.replace(/\n/g, "<br>").replace(/ --([^ ^-])/g, " $1");
+			if (window.is_gm || is_current_player || knows_language) {
+				msg.content = `
+					{{displaymessage=${msg.content}}}
+					{{displaylanguage=(${speech.language})}}
+					{{titlelanguage=${speech.language}}}
+					{{translated=${translated}}}`;
+				msg.rolltemplate = "inlanguage";
+				delete msg.listenerid;
+			}
+		}
+		return {through: true, params};
+	}
+
+	d20plus.chat.outgoing = () => {
+		const tc = d20.textchat.$textarea;
+
+		if (d20plus.cfg.getOrDefault("chat", "emoji")) {
+			tc.val(tc.val().replace(/(:\w*?:)/g, (m0, m1) => {
+				const clean = m1.replace(/:/g, "");
+				return d20plus.chat.emojiIndex && d20plus.chat.emojiIndex[clean] ? `[${clean}](https://github.com/TheGiddyLimit/emoji-dump/raw/master/out/${clean}.png)` : m1;
+			}));
+		}
+
+		let text = tc.val();
+		const srcText = text;
+
+		if (d20plus.cfg.getOrDefault("chat", "commands")) {
+			// add custom commands
+			text = text.replace(/^\/wb(.*?)$/gm, d20plus.chat.sendReply);
+			text = text.replace(/^\/ws(.*?)$/gm, d20plus.chat.sendToSelected);
+			text = text.replace(/^\/ttms( |$)/s, "/talktomyself$1");
+			text = text.replace(/^\/mtms(.*?)$/s, d20plus.chat.sendMyself);
+			text = text.replace(/^\/help(.*?)$/s, d20plus.chat.help);
+			if (d20plus.cfg.getOrDefault("chat", "languages")) text = text.replace(/\/in(.*?)$/gm, d20plus.chat.sendParsedInLanguage);
+		}
+
+		if (d20plus.cfg.getOrDefault("chat", "social")) {
+			const speakingto = $("#speakingto").val();
+			const speakingin = $("#speakingin").val();
+
+			if (speakingin) {
+				if (!text.match(/^\/(.*?)$/)) text = d20plus.chat.sendInLanguage(text, speakingin);
+			}
+
+			if (speakingto && speakingto !== "ttms") {
+				text = text.replace(/^([^/]*?)$/mgu, (...str) => {
+					const prepared = str[1].replace(/\/(r|roll) ([ \dd+-]*?)$/umg, "[[$2]]");
+					return `/w "${speakingto}" ${prepared}`;
+				});
+			}
+		}
+
+		const toSend = $.trim(text);
+		if (text !== srcText && text) d20plus.chat.localHistory.push($.trim(srcText));
+
+		if (d20plus.chat.msgInLang) {
```

Comment:
Might be better not to delete this stuff and remove the chat menu after a sent message, as people will probably want to send multiple language messages in a row. Instead, it should probably be deleted when the language chat menu is collapsed.

Reply:
I think now I've done it right. Added small notification which resets TO/IN when clicked